src/state/slices/common-selectors.ts

```diff
@@ -281,3 +286,48 @@ export const selectIsAssetWithoutMarketData = createSelector(
     return !marketData || marketData.price === '0'
   },
 )
+
+export const selectAssetsBySearchQuery = createCachedSelector(
+  selectAssetsSortedByMarketCap,
+  selectPortfolioAssetBalancesBaseUnit,
+  selectPortfolioUserCurrencyBalances,
+  selectMarketDataUsd,
+  selectSearchQueryFromFilter,
+  (
+    sortedAssets: Asset[],
+    portfolioBalancesCryptoBaseUnit,
+    portfolioBalancesUserCurrency,
+    marketDataUsd,
+    searchQuery?: string,
+  ): Asset[] => {
+    if (!searchQuery) return sortedAssets
+
+    const matchedAssets = matchSorter(sortedAssets, searchQuery ?? '', {
+      keys: ['name', 'symbol', 'assetId'],
+      threshold: matchSorter.rankings.CONTAINS,
+    })
```

Comment:
Seems like we are also prematching in the select asset search, couldn't find a better algo, if someone with a bigger 🧠 comes here